pkg/cluster/deploy.go

```diff
@@ -30,3 +47,92 @@ func (u *simpleUpgrader) Deploy(ctx context.Context, cs *api.OpenShiftManagedClu
 	}
 	return nil
 }
+
+func hashScaleSets(azuretemplate map[string]interface{}) (map[string]string, error) {
+	ssHashes := make(map[string]string)
+	for _, r := range jsonpath.MustCompile("$.resources[?(@.type='Microsoft.Compute/virtualMachineScaleSets')]").Get(azuretemplate) {
+		resource, ok := r.(map[string]interface{})
+		if !ok {
+			continue
+		}
+
+		// cleanup capacity so that no unnecessary VM rotations are going
+		// to occur because of a scale up/down.
+		jsonpath.MustCompile("$.sku.capacity").Delete(resource)
+
+		// hash scale set
+		data, err := json.Marshal(resource)
+		if err != nil {
+			return nil, err
+		}
+		hf := sha256.New()
+		hf.Write(data)
+
+		scaleSetName := jsonpath.MustCompile("$.name").MustGetString(resource)
+		// drop the ss- prefix
+		ssHashes[scaleSetName[3:]] = base64.StdEncoding.EncodeToString(hf.Sum(nil))
+	}
+	return ssHashes, nil
+}
+
+type vmInfo struct {
+	ComputerName string `json:"computerName,omitempty"`
```

Comment:
I think we should use `*VirtualMachineScaleSetVM.Name` to populate this field (example: "ss-master_0"), not `*VirtualMachineScaleSetVM.VirtualMachineScaleSetVMProperties.OsProfile.ComputerName` (example: "master-000000").  This saves us "ss-" prefix trims, base36 shenanigans, reliance on mapping scaleset name and "computername", etc.  I believe we just split on "_".

Reply:
If we are going to use VirtualMachineScaleSetVM.Name it means going back to listing all VMs.